### PR TITLE
feat: change source of packages from local feed to live npm/nuget feed

### LINKED ISSUE #5484: Package Manager: Source packages from npm / nuget feeds
Update the package manager to use the raw npm search and nuget search feeds to source the available packages.

- [x] A developer can list the packages in a package feed, and that list displays summary information about those packages (name, author, version minimum) 
- [x] A developer can choose to install a package into any bot currently open in my bot project 
- [x] Initially, Composer connects to the appropriate package source based on bot language (NuGet or npm)

### PR DESCRIPTION
**Pull request by benbrown**
## Description

Updates the package manager to use the npm and nuget search API rather than a static feed we hand-manage.

This currently uses these feeds:

For NPM, any package tagged `bf-component`:
https://registry.npmjs.org/-/v1/search?text=keywords:bf-component&size=100&from=0

For nuget, any package tagged `bf-component`, including pre-release packages:
https://azuresearch-usnc.nuget.org/query?q=Tags:%22bf-component%22&prerelease=true

Note: These feeds will be updated to search only Microsoft-published packages before we ship.
Also note: A follow-up PR will allow users to add their own feeds, search feeds, etc.

## Task Item

Fixes #5484 

w4chmtH8fi

## To Do

- [x] Update current packages to include `bf-component` keyword/tag
- [x] Update feed to include the real tag
- [x] Update feeds to include namespace?
- [x] Double check error handling

### CODE REVIEW

File: extensions/packageManager/src/client/Library.tsx [hunk 1]
```diff
@@ -146,11 +153,12 @@ const Library: React.FC = () => {
     // find all categories listed in the available libraries
     const categories = [DEFAULT_CATEGORY];
     if (availableLibraries) {
-      const availableCompatibleLibraries = availableLibraries.filter(component => isCompatible(component));
+      const availableCompatibleLibraries = availableLibraries; // .filter(component => isCompatible(component));
```
Comment (a-b-r-o-w-n) [inline extensions/packageManager/src/client/Library.tsx:156]: can the comment be removed? I see `isCompatible` was added to the item below.

File: extensions/packageManager/src/node/index.ts [hunk 2]
```diff
@@ -9,6 +9,41 @@ import { SchemaMerger } from '@microsoft/bf-dialog/lib/library/schemaMerger';
...
+    return feed.data.map((i) => { return {
+      name: i.id,
+      version: i.version,
+      authors: i.authors[0],
+      keywords: i.tags,
+      repository: i.projectUrl,
+      description: i.description,
+      language: 'c#',
+      source: 'nuget',
+    }});
```
Comment (a-b-r-o-w-n) [inline extensions/packageManager/src/node/index.ts:39]: Nit: this is formatted strangely. Can you apply prettier to this (an the ui files) please?

File: extensions/packageManager/src/node/index.ts [hunk 3]
```diff
@@ -58,6 +119,35 @@ export default async (composer: IExtensionRegistration): Promise<void> => {
         recentlyUsed: recentlyUsed,
       });
     },
+    getFeed: async function (req, res) {
+      // read the list of sources from the config file.
```
Comment (a-b-r-o-w-n) [inline extensions/packageManager/src/node/index.ts:123]: I think this comment is here by mistake.

File: extensions/packageManager/src/node/index.ts [hunk 4]
```diff
@@ -58,6 +119,35 @@ export default async (composer: IExtensionRegistration): Promise<void> => {
...
+      const packageSources = [req.query.url];
+
+
+      let combined = [];
+      for (let s = 0; s < packageSources.length; s++) {
```
Comment (a-b-r-o-w-n) [inline extensions/packageManager/src/node/index.ts:128]: Convert this to a `forEach` instead. Or use `for (const s of packageSources)`

**Review by a-b-r-o-w-n — CHANGES_REQUESTED** [review]

File: extensions/packageManager/src/client/Library.tsx (hunk 1 above)
Comment (benbrown) [reply extensions/packageManager/src/client/Library.tsx:156]: yup thanks!

File: extensions/packageManager/src/node/index.ts (hunk 2 above)
Comment (benbrown) [reply extensions/packageManager/src/node/index.ts:39]: I applied the same eslintrc rules from the main composer project. good idea.

File: extensions/packageManager/src/node/index.ts (hunk 3 above)
Comment (benbrown) [reply extensions/packageManager/src/node/index.ts:123]: thx

File: extensions/packageManager/src/node/index.ts (hunk 4 above)
Comment (benbrown) [reply extensions/packageManager/src/node/index.ts:128]: - [x]

File: extensions/packageManager/src/client/Library.tsx [hunk 5]
```diff
@@ -146,11 +153,12 @@ const Library: React.FC = () => {
...
       availableCompatibleLibraries.forEach((item) => {
         if (!item.category) {
           item.category = DEFAULT_CATEGORY;
         }
+        item.isCompatible = isCompatible(item);
         if (item.category && categories.indexOf(item.category) == -1) {
```
[the remark this answers is not in the document]
Comment (benbrown) [reply extensions/packageManager/src/client/Library.tsx:162]: thx

File: extensions/packageManager/src/client/Library.tsx [hunk 6]
```diff
@@ -110,28 +104,41 @@ const Library: React.FC = () => {
...
+        setFeed('nuget');
+      }
+    }
+  },[feeds]);
+
+  useEffect(() => {
+    if (feed && feeds.length) {
+      getLibraries();
+    }
+  },[feed]);
```
[the remark this answers is not in the document]
Comment (benbrown) [reply extensions/packageManager/src/client/Library.tsx:129]: thx

**Comment by benbrown** [comment]
@a-b-r-o-w-n @hatpick thanks for the great feedback. eagle eye as always!

i addressed all the issues you pointed out and one or two others.

File: extensions/packageManager/package.json [hunk 7]
```diff
@@ -55,6 +55,22 @@
...
+    "eslint-plugin-format-message": "6.2.3",
+    "eslint-plugin-import": "2.20.2",
+    "eslint-plugin-jsx-a11y": "6.2.3",
+    "eslint-plugin-lodash": "7.1.0",
+    "eslint-plugin-notice": "0.9.10",
+    "eslint-plugin-prettier": "3.1.3",
+    "eslint-plugin-react": "7.20.0",
+    "eslint-plugin-react-hooks": "4.0.2",
+    "eslint-plugin-security": "1.4.0",
+    "prettier": "2.0.5",
```
Comment (a-b-r-o-w-n) [inline extensions/packageManager/package.json:73]: There is already a prettier config in the extensions root. You can just use vs code to format without having to add all of these dependencies.

Comment (benbrown) [reply extensions/packageManager/package.json]: OK removed!

**Comment by benbrown** [comment]
@a-b-r-o-w-n ready to go i think! want to take one more look?

**Review by a-b-r-o-w-n — APPROVED** [review]

**Pull request merged**